lib/natalie/repl/repl.rb

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/env ruby
+require 'io/console'
+require_relative 'highlight'
+require_relative 'model'
+require_relative 'suggestions'
+require_relative 'constants'
+
+module Natalie
+
+  class StyledString
+    def initialize(str, style)
+      @str = str
+      @style = style
+    end
+
+    def to_s
+      "#{@style}#{@str}#{RESET_STYLE_ASCCI_CODE}"
+    end
+
+    def length
+      @str.length
+    end
+  end
+
+  class GenericRepl
+    def initialize(vars)
+      @highlighters = [
+        Natalie::KEYWORD_HIGHLIGHT,
+        Natalie::PASCAL_CASE_HIGLIGHT,
+        Natalie::CAMEL_CASE_HIGLIGHT,
+      ]
+      @model = Natalie::ReplModel.new
+      @suggestions = Natalie::SuggestionProvider.new(vars)
+      reset 
+    end
+
+    def hello_natalie
+      puts "\u001b[32m
+    @@@@@@@@@@@@@@@@@@@
+    @                 @
+    @   N         N   @     AA     TTTTTTTTT     AA     L       I    EEEEE
+    @   N N       N   @    A  A        T        A  A    L       I    E
+    @   N N N     N   @   A    A       T       A    A   L       I    EEE
+    @   N     N   N   @   AAAAAA       T       AAAAAA   L       I    E
+    @   N       N N   @   A    A       T       A    A   LLLLL   I    EEEEE
+    @                 @ 
+    @@@@@@@@@@@@@@@@@@@
+    \ \ \ \ \ \ \ :::
+    @@@@@@@@@@@@@@@@@@@#{RESET_STYLE_ASCCI_CODE}
+    \n
+    "
```

Comment:
What do you think about using those braille characters to generate something a bit more like the actual logo...

```
  ⡎⠉⣉⣉⣉⣉⣉⣉⣉⣉⠉⢱
  ⡇⢸ ⡢⡢⡀ ⠠⡢ ⡇⢸
  ⡇⢸ ⡪⡊⠪⡢⡨⡪ ⡇⢸
  ⡇⢸ ⠊⠂ ⠈⠊⠊ ⡇⢸
  ⣇⣀⣉⣉⣉⣉⣉⣉⣉⣉⣀⣸
⡔⣒⠒⠒⠒⠒⠚⠒⠒⠓⠒⠒⠒⠒⠒⢢
⢇⣉⣀⣀⣀⣀⣀⣀⣀⣀⣁⣉⣉⣉⣁⡸
```

Note: GitHub has some alignment issues here, but this seems to work in the terminal for me. Would love to see if it works in your terminal too.

Reply:
it's wonderful and it's all I ever wanted :^) Out of curiosity, how long did it take?

Reply:
Didn't take too long. I used this tool: https://braille-txt.github.io/ :wink:

Reply:
This is cheating *coff coff*